test/es-module/test-esm-import-flag.mjs

```diff
@@ -0,0 +1,168 @@
+import { spawnPromisified } from '../common/index.mjs';
+import fixtures from '../common/fixtures.js';
+import assert from 'node:assert';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+const cjsEntry = fixtures.path('es-modules', 'cjs-file.cjs');
+const cjsImport = fixtures.fileURL('es-modules', 'cjs-file.cjs');
+const mjsEntry = fixtures.path('es-modules', 'mjs-file.mjs');
+const mjsImport = fixtures.fileURL('es-modules', 'mjs-file.mjs');
+
+
+describe('import modules using --import', { concurrency: true }, () => {
+  it('should import when using --eval', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        '--eval', 'console.log("log")',
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\nlog\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import when using --eval and --input-type=module', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        '--input-type', 'module',
+        '--eval', 'console.log("log")',
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\nlog\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import when main entrypoint is a cjs file', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
```

Comment:
Does this effectively replace `--require` then?

Reply:
yes but with side effects - it forces esm to run/load